src/config.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""Spark Configuration Hub configuration."""
+
+import re
+from typing import Optional
+
+from models import S3ConnectionInfo
+from utils import WithLogging
+
+
+class SparkConfigurationHubConfig(WithLogging):
+    """Spark Confiuration Hub Configuration."""
+
+    _ingress_pattern = re.compile("http://.*?/|https://.*?/")
+
+    def __init__(
+        self,
+        s3_connection_info: Optional[S3ConnectionInfo],
+    ):
+        self.s3_connection_info = s3_connection_info
+
+    _base_conf: dict[str, str] = {
+        "spark.hadoop.fs.s3a.connection.ssl.enabled": "false",
+        "spark.hadoop.fs.s3a.path.style.access": "true",
+        "spark.eventLog.enabled": "true",
```

Comment:
**todo** yes, this should probably go in the s3 config as well, as Bikalpa mentioned